architecture/consistency.md

```diff
@@ -0,0 +1,106 @@
+---
+title: Consistency
+summary: Achieving consistency in distributed, message-driven systems
+reviewed: 2023-07-18
+callsToAction: ['solution-architect']
+redirects:
+- nservicebus/azure/understanding-transactionality-in-azure
+- nservicebus/understanding-transactions-in-windows-azure
+---
+
+## Transactions
+
+Transactions are designed to keep a system in a consistent state at any point in time. This is achieved by ensuring interdependent operations are completed as a single [unit of work](https://en.wikipedia.org/wiki/Unit_of_work). If any single operation fails, then the entire transaction fails and the system is reverted to the state it was in before the transaction started.
+
+### Limitations
+
+There are several challenges and limitations when relying on transactions:
+
+#### Scalability
+
+Scalable systems need to handle high levels of concurrency. Transactions need concurrency control mechanisms to ensure proper isolation between transactions trying to access the same resources. These concurrency control mechanisms can severely impact a resource's performance and introduce bottlenecks that complicate scaling.
+
+When using [horizontal scaling](https://learn.microsoft.com/en-us/azure/well-architected/scalability/design-scale#understand-horizontal-and-vertical-scaling), additional mechanisms are required to ensure the desired level of consistency across all scale units. This typically involves complicated distributed consensus algorithms. Depending on the consistency guarantees required, many, or even all, nodes of a scaled component need to participate in the transaction, negatively impacting latency and availability.
+
+#### Cost
+
+Besides the performance impact of transactions, transactional operations may even be charged extra on certain managed cloud services. Broad usage of transactional consistency may therefore incur significant costs. It is important to understand a cloud vendor's pricing models to estimate the financial cost of using transactions.
+
+#### Timeouts
+
+Transactions impact concurrent operations on a locked resource. This may block other actors and consume additional resources. Therefore, transactions should be completed within a short time window to avoid issues. Cloud services typically restrict shorter transaction lifetimes than traditional on-premises technologies.
```

Comment:
```suggestion
Transactions impact concurrent operations on a locked resource. This may block other actors and consume additional resources. Therefore, transactions should be completed within a short time window to avoid issues. Cloud services typically restrict transaction lifetimes more than traditional on-premises technologies.
```